app/models/instance_filter.rb

```diff
@@ -40,12 +46,33 @@ def scope_for(key, value)
     end
   end
 
+  def status_scope(value)
+    # The `join where` queries here look like they have a bug due to `domain_block`
+    # vs `domain_blocks`, however the table is `domain_blocks` while the relation on
+    # Instances is `domain_block`
+    case value.to_sym
+    when :allowed
+      Instance.joins(:domain_allow).reorder(Arel.sql('domain_allows.id desc'))
+    when :suspended
+      Instance.joins(:domain_block).where(domain_blocks: { severity: :suspend }).reorder(Arel.sql('domain_blocks.id desc'))
+    when :limited
+      Instance.joins(:domain_block).where(domain_blocks: { severity: :silence }).or(Instance.joins(:domain_block).where(domain_blocks: { severity: :noop })).reorder(Arel.sql('domain_blocks.id desc'))
```

Comment:
I think this is better written as:
```suggestion
    when :limited
      Instance.joins(:domain_block).where(domain_blocks: { severity: [:silence, :noop] }).reorder(Arel.sql('domain_blocks.id desc'))
```

Relatedly, I'm not sure “limited” is the correct moderator-facing term here given that it's not just about the “limited” severity but also other restrictions… that being said, this is not a new issue.

Reply:
Yeah, it was either group these under "limited" or have a separate group for them. After earlier feedback I changed to group under limited, since each row shows why & how it's limited